src/DataBuilder.php

```diff
@@ -977,34 +976,73 @@ public function getCustomDataMethod()
         return $this->customDataMethod;
     }
 
-    protected function getCustomForPayload($toLog, $context)
+    /**
+     * Returns the processed custom value to be sent with the payload. If there
+     * is no custom value an empty array is returned.
+     *
+     * @param Throwable|string $toLog
+     * @param array            $context
+     *
+     * @return array
+     */
+    protected function getCustomForPayload(Throwable|string $toLog, array $context): array
     {
-        $custom = $this->getCustom();
+        $custom = $this->resolveCustomContent($this->getCustom());
 
-        // Make this an array if possible:
-        if ($custom instanceof \Serializable) {
-            $custom = $custom->serialize();
-        } elseif (is_null($custom)) {
-            $custom = array();
-        } elseif (!is_array($custom)) {
-            $custom = get_object_vars($custom);
-        }
-        
         if ($customDataMethod = $this->getCustomDataMethod()) {
-            $customDataMethodContext = isset($context['custom_data_method_context']) ?
-                $context['custom_data_method_context'] :
-                null;
-                
+            $customDataMethodContext = $context['custom_data_method_context'] ?? null;
+
             $customDataMethodResult = $customDataMethod($toLog, $customDataMethodContext);
-            
+
             $custom = array_merge($custom, $customDataMethodResult);
         }
-        
+
         unset($context['custom_data_method_context']);
 
         return $custom;
     }
-    
+
+    /**
+     * This method transforms $custom into an array that can be processed and sent in the payload.
+     *
+     * @param mixed $custom
+     *
+     * @return array
+     * @throws \Exception If Serializable::serialize() returns an invalid type an exception can be thrown.
+     *                    This can be removed once support for Serializable has been removed.
+     */
+    protected function resolveCustomContent(mixed $custom): array
```

Comment:
This method was added so that we could simplify/consolidate the logic around transforming the `custom` parameter into an array. 

I tossed around the idea of making this method recursive. So, it could handle a nested object or multidimensional array. However, recursion could cause a lot of issues if we ran into circular references. I also don't know if we even support multidimensional data on the Rollbar server processing side. After I had thought about it a bit. I decided recursion is not a good idea here. 

This is partly why I also deprecated the use of an object that does not implement the new `Rollbar\SerializerInterface` interface. We probably should not just be trying to serialize someone else's object without them having a very clear understanding of what will be included. This should be done by them explicitly including it in the `serialize()` method.